tldw_Server_API/tests/Evaluations/test_recipe_runs_repository.py

```
@@ -0,0 +1,113 @@
+
+def test_set_recipe_run_children_rejects_missing_parent(tmp_path) -> None:
+    db = EvaluationsDatabase(str(tmp_path / "evaluations.db"))
+
+    with pytest.raises(ValueError, match="parent recipe run does not exist"):
+        db.set_recipe_run_children("missing-parent", ["child-a"])
+
+
+def test_create_recipe_run_rolls_back_when_child_links_fail(tmp_path) -> None:
+    db = EvaluationsDatabase(str(tmp_path / "evaluations.db"))
+
+    with pytest.raises(Exception):
```

Ruff (0.15.7)

[warning] Do not assert blind exception: `Exception`

(B017)